internal/provider/resource_nsx_edge_cluster.go

```diff
@@ -0,0 +1,269 @@
+			},
+			"high_availability": {
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  "One among: ACTIVE_ACTIVE, ACTIVE_STANDBY",
+				ValidateFunc: validation.StringInSlice([]string{"ACTIVE_ACTIVE", "ACTIVE_STANDBY"}, false),
+			},
+			"mtu": {
+				Type:         schema.TypeInt,
+				Required:     true,
+				Description:  "Maximum transmission unit size for the cluster",
+				ValidateFunc: validation.IntBetween(1600, 9000),
+			},
+			"asn": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Description:  "ASN for the cluster",
+				ValidateFunc: validation.IntBetween(1, int(math.Pow(2, 32))),
+			},
+			"skip_tep_routability_check": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "Set to true to bypass normal ICMP-based check of Edge TEP / host TEP routability (default is false, meaning do check)",
+				Default:     false,
+			},
+			"tier1_unhosted": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "Select whether Tier-1 being created per this spec is hosted on the new Edge cluster or not (default value is false, meaning hosted)",
+				Default:     false,
+			},
+			"internal_transit_subnets": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "Subnet addresses in CIDR notation that are used to assign addresses to logical links connecting service routers and distributed routers",
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+			"transit_subnets": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "Transit subnet addresses in CIDR notation that are used to assign addresses to logical links connecting Tier-0 and Tier-1s",
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+			"edge_node": {
+				Type:        schema.TypeList,
+				Required:    true,
+				Description: "The nodes in the edge cluster",
+				Elem:        nsx_edge_cluster.EdgeNodeSchema(),
+			},
+		},
+	}
+}
+
+func resourceNsxEdgeClusterCreate(ctx context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*api_client.SddcManagerClient).ApiClient
+
+	postParams := &nsxt_edge_clusters.CreateEdgeParams{
+		EdgeCreationSpec: nsx_edge_cluster.GetNsxEdgeClusterCreationSpec(data),
+		Context:          ctx,
+	}
+
+	_, task, err := client.NSXTEdgeClusters.CreateEdge(postParams.WithTimeout(constants.DefaultVcfApiCallTimeout))
+
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	tflog.Info(ctx, "Edge cluster creation has started.")
+	err = meta.(*api_client.SddcManagerClient).WaitForTaskComplete(ctx, task.Payload.ID, false)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	getParams := &nsxt_edge_clusters.GetEdgeClustersParams{}
+	var clusters *nsxt_edge_clusters.GetEdgeClustersOK
+	clusters, err = client.NSXTEdgeClusters.GetEdgeClusters(getParams.WithTimeout(constants.DefaultVcfApiCallTimeout))
```

Comment:
The task response from client.NSXTEdgeClusters.CreateEdge contains "Resources []*Resource" upon filtering that slice of Resource you will, likely, be able to find the newly created NSX Edge Cluster ID without an additional request.

See: resourceDomainCreate for example usage of GetResourceIdAssociatedWithTask, that will extract the ID of the resulting resource for you

Reply:
Good suggestion, I tested this again and unfortunately the task and all 61 subtasks have their resources array set to nil. This looks like a deficiency in the API.
The good news is that there is a requirement for cluster names to be unique so we can safely identify the cluster by its name. I will leave a comment in the code to make sure that this is not regarded as good practice